### PR TITLE
Add session-cookie-keywords

### PR DESCRIPTION
**Pull request by MartinKirchner**
Add option to configure the cookie options in more detail.
This might be necessary to let the backend server access the affinity cookie
or to turn off the `httponly` flag.

The default value is still `indirect nocache httponly` to be backwards
compatible.

### CODE REVIEW

**Comment by MartinKirchner** [comment]
Closes #566

File: docs/content/en/docs/configuration/keys.md [hunk 1]
```diff
@@ -198,6 +198,7 @@ The table below describes all supported configuration keys.
 | [`session-cookie-name`](#affinity)                   | cookie name                             | Backend |                    |
 | [`session-cookie-shared`](#affinity)                 | [true\|false]                           | Backend | `false`            |
 | [`session-cookie-strategy`](#affinity)               | [insert\|prefix\|rewrite]               | Backend |                    |
+| [`session-cookie-keywords`](#affinity)               | cookie options                          | Backend | `indirect nocache httponly`     |
```
Comment (jcmoraisjr) [inline docs/content/en/docs/configuration/keys.md:201]: sort alphabetically

File: docs/content/en/docs/configuration/keys.md [hunk 2]
```diff
@@ -336,14 +337,15 @@ See also:
...
-| `session-cookie-strategy`   | `Backend` | `insert`        |       |
+| Configuration key           | Scope     | Default                     | Since |
+|-----------------------------|-----------|-----------------------------|-------|
+| `affinity`                  | `Backend` | `false`                     |       |
+| `cookie-key`                | `Global`  | `Ingress`                   |       |
+| `session-cookie-dynamic`    | `Backend` | `true`                      |       |
+| `session-cookie-name`       | `Backend` | `INGRESSCOOKIE`             |       |
+| `session-cookie-shared`     | `Backend` | `false`                     | v0.8  |
+| `session-cookie-strategy`   | `Backend` | `insert`                    |       |
+| `session-cookie-keywords`   | `Backend` | `indirect nocache httponly` | v0.11 |
```
Comment (jcmoraisjr) [inline docs/content/en/docs/configuration/keys.md:348]: sort alphabetically as well

File: docs/content/en/docs/configuration/keys.md [hunk 3]
```diff
@@ -353,6 +355,7 @@ Configure if HAProxy should maintain client requests to the same backend server.
 * `session-cookie-strategy`: the cookie strategy to use (insert, rewrite, prefix). `insert` is the default value if not declared.
 * `session-cookie-shared`: defines if the persistence cookie should be shared between all domains that uses this backend. Defaults to `false`. If `true` the `Set-Cookie` response will declare all the domains that shares this backend, indicating to the HTTP agent that all of them should use the same backend server.
 * `session-cookie-dynamic`: indicates whether or not dynamic cookie value will be used. With the default of `true`, a cookie value will be generated by HAProxy using a hash of the server IP address, TCP port, and dynamic cookie secret key. When `false`, the server name will be used as the cookie name. Note that setting this to `false` will have no impact if [use-resolver](#dns-resolvers) is set.
+* `session-cookie-keywords`: options to the `cookie` option like `nocache`, `preserve`, `httponly`. `indirect nocache httponly` is the default if not declared to be backwards compatible to earlier versions.
```
Comment (jcmoraisjr) [inline docs/content/en/docs/configuration/keys.md:358]: add a reference to the default value only if `insert` strategy is used.

try also to rephrase the last sentence to something like "the default is ..." in order to separate the former list of keywords to the later one.

File: pkg/converters/ingress/types/annotations.go [hunk 4]
```diff
@@ -110,6 +110,7 @@ const (
 	BackSessionCookieName      = "session-cookie-name"
 	BackSessionCookieShared    = "session-cookie-shared"
 	BackSessionCookieStrategy  = "session-cookie-strategy"
+	BackSessionCookieKeywords  = "session-cookie-keywords"
```
Comment (jcmoraisjr) [inline pkg/converters/ingress/types/annotations.go:113]: sort alphabetically

File: docs/content/en/docs/configuration/keys.md (hunk 1 above)
Comment (MartinKirchner) [reply docs/content/en/docs/configuration/keys.md]: Done

File: docs/content/en/docs/configuration/keys.md (hunk 2 above)
Comment (MartinKirchner) [reply docs/content/en/docs/configuration/keys.md]: Done

File: docs/content/en/docs/configuration/keys.md (hunk 3 above)
Comment (MartinKirchner) [reply docs/content/en/docs/configuration/keys.md]: Done

File: pkg/converters/ingress/types/annotations.go (hunk 4 above)
Comment (MartinKirchner) [reply pkg/converters/ingress/types/annotations.go]: Done

**Comment by jcmoraisjr** [comment]
Merging, thanks!

**Pull request merged**